applications/GeoMechanicsApplication/custom_elements/geo_steady_state_Pw_piping_element.h

```diff
@@ -118,12 +118,16 @@ class KRATOS_API(GEO_MECHANICS_APPLICATION) GeoSteadyStatePwPipingElement : publ
         Element::Initialize(rCurrentProcessInfo);
         // all these except the PIPE_ELEMENT_LENGTH seem to be in the erosion_process_strategy only. Why do this: it is used in output for dGeoFlow
         this->SetValue(PIPE_ELEMENT_LENGTH, CalculateLength(this->GetGeometry()));
-        this->SetValue(PIPE_EROSION, false);
-        constexpr double small_pipe_height = 1e-10;
-        this->SetValue(PIPE_HEIGHT, small_pipe_height);
-        this->SetValue(PREV_PIPE_HEIGHT, small_pipe_height);
-        this->SetValue(DIFF_PIPE_HEIGHT, 0.);
-        this->SetValue(PIPE_ACTIVE, false);
+
+        if (!mIsInitialized) {
+            mIsInitialized = true;
+            this->SetValue(PIPE_EROSION, false);
+            constexpr double small_pipe_height = 1e-10;
+            this->SetValue(PIPE_HEIGHT, small_pipe_height);
+            this->SetValue(PREV_PIPE_HEIGHT, small_pipe_height);
+            this->SetValue(DIFF_PIPE_HEIGHT, 0.);
+            this->SetValue(PIPE_ACTIVE, false);
+        }
```

Comment:
It's perhaps slightly more logical to first carry out the initialization, and only then set the flag that the element has been initialized. But this could also be my own preference. What do you think?
```suggestion
        if (!mIsInitialized) {
            this->SetValue(PIPE_EROSION, false);
            constexpr double small_pipe_height = 1e-10;
            this->SetValue(PIPE_HEIGHT, small_pipe_height);
            this->SetValue(PREV_PIPE_HEIGHT, small_pipe_height);
            this->SetValue(DIFF_PIPE_HEIGHT, 0.);
            this->SetValue(PIPE_ACTIVE, false);
            mIsInitialized = true;
        }
```

Reply:
I get your reasoning and I don't have a strong preference, moved!